src/univariate/continuous/normalcanon.jl

```diff
@@ -1,24 +1,28 @@
 ## Canonical Form of Normal distribution
 
-immutable NormalCanon <: ContinuousUnivariateDistribution
-    η::Float64       # σ^(-2) * μ
-    λ::Float64    # σ^(-2)
-    μ::Float64       # μ
+immutable NormalCanon{T<:Real, S<:Real} <: ContinuousUnivariateDistribution
```

Comment:
Why allow for mixed precision here? It seems more complicated than really useful.